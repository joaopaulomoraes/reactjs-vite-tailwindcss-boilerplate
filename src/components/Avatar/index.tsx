import { classNames } from 'utils'

type Size = 'small' | 'medium' | 'large'

type AvatarProps = {
  size?: Size
  src?: string
  alt?: string
}

const sizes: Record<Size, string> = {
  small: 'w-10 h-10',
  medium: 'w-12 h-12',
  large: 'w-14 h-14',
}

const EmptyAvatar = ({ size = 'medium' }: Pick<AvatarProps, 'size'>) => (
  <span
    data-testid='empty-avatar'
    className={classNames('inline-block overflow-hidden bg-gray-100 rounded-full', sizes[size])}
  >
    <svg className='h-full w-full text-gray-300' fill='currentColor' viewBox='0 0 24 24'>
      <path d='M24 20.993V24H0v-2.996A14.977 14.977 0 0112.004 15c4.904 0 9.26 2.354 11.996 5.993zM16.002 8.999a4 4 0 11-8 0 4 4 0 018 0z' />
    </svg>
  </span>
)

export default function Avatar({ size = 'medium', src, alt }: AvatarProps) {
  if (!src) {
    return <EmptyAvatar size={size} />
  }

  return (
    <img className={classNames('inline-block rounded-full', sizes[size])} src={src} alt={alt} />
  )
}
