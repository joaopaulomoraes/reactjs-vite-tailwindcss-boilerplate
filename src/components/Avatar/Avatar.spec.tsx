import { render, screen } from '@testing-library/react'

import Avatar from '.'

describe('<Avatar />', () => {
  const props = {
    src: 'https://gravatar.com/4405735f6f3129e0286d9d43e7b460d0',
    alt: 'Avatar',
  }

  it('should render the medium Avatar as default', async () => {
    const { container } = render(<Avatar {...props} />)

    expect(screen.getByRole('img', { name: /Avatar/i })).toBeInTheDocument()

    expect(container.firstChild).toHaveClass('inline-block w-12 h-12 rounded-full')

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the small Avatar', () => {
    const { container } = render(<Avatar size='small' {...props} />)

    expect(screen.getByRole('img', { name: /Avatar/i })).toBeInTheDocument()

    expect(container.firstChild).toHaveClass('inline-block w-10 h-10 rounded-full')

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the medium Avatar', () => {
    const { container } = render(<Avatar size='medium' {...props} />)

    expect(screen.getByRole('img', { name: /Avatar/i })).toBeInTheDocument()

    expect(container.firstChild).toHaveClass('inline-block w-12 h-12 rounded-full')
  })

  it('should render the large Avatar', () => {
    const { container } = render(<Avatar size='large' {...props} />)

    expect(screen.getByRole('img', { name: /Avatar/i })).toBeInTheDocument()

    expect(container.firstChild).toHaveClass('inline-block w-14 h-14 rounded-full')

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the empty Avatar', () => {
    const { container } = render(<Avatar />)

    expect(screen.getByTestId('empty-avatar')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
