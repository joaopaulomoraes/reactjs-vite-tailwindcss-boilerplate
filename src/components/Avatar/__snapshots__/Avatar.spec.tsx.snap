// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Avatar /> > should render the empty Avatar 1`] = `
<span
  class="inline-block overflow-hidden bg-gray-100 rounded-full w-12 h-12"
  data-testid="empty-avatar"
>
  <svg
    class="h-full w-full text-gray-300"
    fill="currentColor"
    viewBox="0 0 24 24"
  >
    <path
      d="M24 20.993V24H0v-2.996A14.977 14.977 0 0112.004 15c4.904 0 9.26 2.354 11.996 5.993zM16.002 8.999a4 4 0 11-8 0 4 4 0 018 0z"
    />
  </svg>
</span>
`;

exports[`<Avatar /> > should render the large Avatar 1`] = `
<img
  alt="Avatar"
  class="inline-block rounded-full w-14 h-14"
  src="https://gravatar.com/4405735f6f3129e0286d9d43e7b460d0"
/>
`;

exports[`<Avatar /> > should render the medium Avatar as default 1`] = `
<img
  alt="Avatar"
  class="inline-block rounded-full w-12 h-12"
  src="https://gravatar.com/4405735f6f3129e0286d9d43e7b460d0"
/>
`;

exports[`<Avatar /> > should render the small Avatar 1`] = `
<img
  alt="Avatar"
  class="inline-block rounded-full w-10 h-10"
  src="https://gravatar.com/4405735f6f3129e0286d9d43e7b460d0"
/>
`;
